### DISCUSSION TITLE
Using BioAge for urine derived biomarkers

### DISCUSSION THREAD

**Discussion by emilijar96** [dayoonkwon/BioAge#43] [Q&A]
Hello there,

I heard about this package and I would really like to try it in a study, but I am not sure if it can be used with biomarkers derived from urine or is it restricted to those from the blood?

Thanks a lot in advance!

Emilija

**Comment by dayoonkwon** [comment]
The `_nhanes` functions are limited to biomarkers available in the NHANES datasets, which are blood-based. However, the `_calc` functions (`kdm_calc`, `phenoage_calc`, `hd_calc`) accept any numeric biomarker — including urine-derived ones — as long as you provide your own training data. The algorithms are agnostic to the source of the biomarker; they just need to correlate with age.
